src/java.base/share/native/libjava/jni_util.c

```diff
@@ -39,7 +39,7 @@
  * negative, or the size is INT_MAX as the macro adds 1
  * that overflows into negative value.
  */
-#define MALLOC_MIN4(len) ((unsigned)(len) >= INT_MAX ? \
+#define MALLOC_MIN4(len) (len >= INT_MAX || len < 0 ? \
```

Comment:
Should this keep the parentheses around `len` for explicit order of operations in case `len` is an expression?

```suggestion
#define MALLOC_MIN4(len) ((len) >= INT_MAX || (len) < 0 ? \
```

Reply:
Good point. Suggestion incorporated